spec/factories/claims.rb

```diff
@@ -56,12 +56,16 @@
 
     trait :approved do
       submitted
-      association(:decision, factory: [:decision, :approved], strategy: :build)
+      after(:create) do |claim|
+        create(:decision, :approved, claim: claim)
```

Comment:
We should avoid calling `create` in factories, otherwise it becomes impossible to instantiate an instance without it being written to the database:

```suggestion
      after(:build) do |claim|
        build(:decision, :approved, claim: claim)
```

Reply:
Actually, you may need to assign it to the claim if the record isn't being saved... Something like this may work:

```suggestion
      after(:build) do |claim|
        claim.decisions.build(attributes_for(:decision, :approved))
```

Reply:
I've gone with `claim.decisions.append(build(:decision, :rejected, claim: claim))`. `claim.decisions.build` complains that `decisions is invalid`, which as far as I can tell is because it is returning a single `decision` rather than a list. There is no `build_list` available.